RustLeanModels/Basic.lean

```diff
@@ -120,195 +140,179 @@ def monotone (l: List Nat) := match l with
   |[] => true
   |h::t => (∀ m ∈ t, h ≤ m) ∧ monotone t
 
-lemma map_sub_monotone: monotone l → monotone (List.map (fun x => x -k ) l) :=by
+theorem map_sub_monotone: monotone l → monotone (List.map (fun x => x -k ) l) :=by
```

Comment:
Optional: In proofs by induction, use `case nil` `case cons ...` to make things more readable?

```
theorem map_sub_monotone: monotone l → monotone (List.map (fun x => x - k) l) := by
  induction l
  case nil =>
    simp
  case cons y l' ih =>
    simp [monotone]
    intro le mon
    simp only [ih mon, and_true]
    intro a ha
    exact Nat.sub_le_sub_right (le a ha) k
```

Reply:
I updated all the induction proofs to make the cases explicit.